modules/c++/cli/source/ArgumentParser.cpp

```diff
@@ -249,11 +249,9 @@ cli::Results* cli::ArgumentParser::parse(const std::vector<std::string>& args)
     std::map<std::string, Argument*> longOptionsFlags;
     std::vector<Argument*> positionalArgs;
 
-    // first, validate the arguments
-    for (mem::VectorOfSharedPointers<cli::Argument>::const_iterator it =
-            mArgs.begin(); it != mArgs.end(); ++it)
+    for (const auto argument : mArgs)
```

Comment:
I should have clarified... for this particular codebase, we can't use any C++11 or newer features (yet) so this will have to get reverted back.